crates/sequencer/src/state.rs

```diff
@@ -143,6 +170,27 @@ impl BlockifierStateReader for &mut State {
     }
 }
 
+impl From<&SerializableState> for State {
+    fn from(serializable_state: &SerializableState) -> Self {
+        let mut state = Self {
+            classes: FxHashMap::default(),
+            compiled_class_hashes: serializable_state.compiled_classes_hash.clone(),
+            contracts: serializable_state.contracts.clone(),
+            storage: serializable_state.storage.clone(),
+            nonces: serializable_state.nonces.clone(),
+        };
+
+        serializable_state.classes.iter().for_each(|(class_hash, contract_class)|{
```

Comment:
why do we need to do this?

Reply:
The `contract_class` is of type `SerializableClassRecord`, hence we need a conversion from this to `ContractClass`  
{ which is not seriaizable, and hence we have a seriaizable type for it } which is what we store in a normal state of the seqeuncer.

<img width="585" alt="image" src="https://github.com/kkrt-labs/ef-tests/assets/41180869/c432ef5a-e2a7-4507-a980-7984db186136">